addon/mixins/routes/application.js

```diff
@@ -31,6 +31,13 @@ export default Ember.Mixin.create({
         outlet: outlet,
         parentView: parentView
       });
+
+      // Check if another modal exists then give it focus
+      Ember.run.later(function() {
```

Comment:
Thanks for the explanation of `Ember.run`. I am pretty new to ember and i didn't knew that. In my work we use angular (u_u) but i use ember in personal projects and freelance.